### PR TITLE
 refactor: refactored into individual submodules

### PR DESCRIPTION
**Pull request by Soaib024**
### Description

Refactored the observability module into individual components 

It can still be consumed as before which will create an Activity Tracker, LogDNA and Sysdig at once, now an option to provision any one of these at a time by referring to submodules have been added

**Changes are backward compatible** 

### Types of changes in this PR
https://github.com/terraform-ibm-modules/terraform-ibm-observability-instances/issues/212

#### No release required

- [ ] Examples or tests (addition or updates of examples or tests)
- [x] Documentation update
- [ ] CI-related update (pipeline, etc.)
- [ ] Other changes that don't affect Terraform code
- [x] Refactor 

#### Release required

- [ ] Bug fix (patch release (`x.x.X`): Change that fixes an issue and is compatible with earlier versions)
- [ ] New feature (minor release (`x.X.x`): Change that adds functionality and is compatible with earlier versions)
- [ ] Breaking change (major release (`X.x.x`): Change that is likely incompatible with previous versions)

##### Release notes content

Replace this text with information that users need to know about the bug fixes, features, and breaking changes. This information helps the merger write the commit message that is published in the release notes for the module.

---

### Checklist for reviewers

- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### Merge actions for mergers

- Merge by using "Squash and merge".
- Use a relevant [conventional commit](https://www.conventionalcommits.org/) message that is based on the PR contents and any release notes provided by the PR author.

    The commit message determines whether a new version of the module is needed, and if so, which semver increment to use (major, minor, or patch).

### CODE REVIEW

File: version.tf [hunk 1]
```diff
@@ -2,10 +2,6 @@ terraform {
   required_version = ">= 1.0.0"
   required_providers {
     # Use "greater than or equal to" range in modules
-    ibm = {
-      source  = "ibm-cloud/ibm"
-      version = ">= 1.49.0"
-    }
     logdna = {
```
Comment (shemau) [inline version.tf:5]: This is required, it is included in the documentation to expose the required version, it also implicitly required by all the modules.

https://github.com/terraform-linters/tflint-ruleset-terraform/blob/v0.2.1/docs/rules/terraform_unused_required_providers.md

**Review by shemau — CHANGES_REQUESTED** [review]
https://developer.hashicorp.com/terraform/language/modules/develop/structure suggests that sub modules are stored in a modules/ subdirectory. The presence of the README is also significant, if a module is considered internal, no README should be provided.

Would it be neater, now that there is a (sub-)module for logdna, activity tracker, and sysdig to have those modules exclude logdna_, at_, activity_tracker and sysdig_ from the variable names and output names. These clarifiers only have a purpose in the root module.

**Comment by Soaib024** [comment]
I have changed those variables name except activity_tracker_provision logdna_provision  and sysdig_provision  in submodules, I think having variable name as provision will not make much sense, also as submodule can be consumed individually we can keep those readmes as it won't cause harm anyway

**Review by shemau — DISMISSED** [review]

**Review by shemau — APPROVED** [review]
LGTM

**Pull request merged**